include/ttmlir/Target/Utils/MLIRToFlatbuffer.h

```diff
@@ -108,6 +107,23 @@ inline ::tt::target::Arch toFlatbuffer(FlatbufferObjectCache &, ArchAttr arch) {
   }
 }
 
+// Overloaded function for DataType
+inline ::tt::target::DataType toFlatbuffer(FlatbufferObjectCache &,
+                                           DataType dtype) {
```

Comment:
It seems like this function just introduces an extra level of indirection, isn't this what the function on line 56 accomplished?

Reply:
There needs to be 2 toFlatbuffer implementations, one for DataType and the other for DataTypeAttr. They both call convertDataType (which used to be original toFlatBuffer before supported_data_types was introduced).
 
Example toFlatbuffer call that uses DataType is around line 370:
```
return ::tt::target::CreateMemoryDescDirect(
      *cache.fbb, &shape, &tileShape, toFlatbuffer(cache, dtype),
      toFlatbuffer(
          cache,
          mlir::cast<MemorySpaceAttr>(memref.getMemorySpace()).getValue()),
      size);
```